plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/action/NewPackageAction.java

```diff
@@ -41,13 +42,37 @@
  */
 @Singleton
 public class NewPackageAction extends AbstractNewResourceAction {
+
+    private final AppContext appContext;
     private final InputValidator nameValidator = new NameValidator();
 
     @Inject
-    public NewPackageAction(JavaResources javaResources, JavaLocalizationConstant localizationConstant) {
+    public NewPackageAction(JavaResources javaResources,
+                            JavaLocalizationConstant localizationConstant,
+                            AppContext appContex    ) {
         super(localizationConstant.actionNewPackageTitle(),
               localizationConstant.actionNewPackageDescription(),
               javaResources.packageIcon());
+        this.appContext = appContex;
+    }
+
+    @Override
+    public void updateInPerspective(@NotNull ActionEvent e) {
+        CurrentProject project = appContext.getCurrentProject();
```

Comment:
You can simply create shared abstract class for New package and new java resource and do work with visibility set in it.